lib/FrameManager.js

```diff
@@ -566,8 +566,15 @@ class Frame {
    * @return {!Promise}
    */
   waitFor(selectorOrFunctionOrTimeout, options = {}, ...args) {
-    if (helper.isString(selectorOrFunctionOrTimeout))
-      return this.waitForSelector(/** @type {string} */(selectorOrFunctionOrTimeout), options);
+    const xpathPattern = /^\/\//;
+
+    if (helper.isString(selectorOrFunctionOrTimeout)) {
+      const string = /** @type {string} */ (selectorOrFunctionOrTimeout);
+      if (string.match(xpathPattern))
```

Comment:
Seeing as we are using an imperfect heuristic, I think we can be more aggressive and say its an XPath if it just starts with `/`. `string.startsWith('/') `.

Reply:
alright, SGTM